gwcs/wcs.py

```diff
@@ -794,6 +794,132 @@ def to_fits_sip(self, bounding_box=None, max_pix_error=0.25, degree=None,
         hdr['cd2_2'] = cdmat[1][1]
         return hdr
 
+    def to_fits_tab(self, bounding_box=None, bin_ext_name='WCS-TABLE',
+                    crd_col_name='coordinates', sampling=1):
+        """
+        Construct a FITS WCS ``-TAB``-based approximation to the WCS
+        in the form of a FITS header and a binary table extension. For the
+        description of the FITS WCS ``-TAB`` convention, see
+        "Representations of spectral coordinates in FITS" in
+        `Greisen, E. W. et al. A&A 446 (2) 747-771 (2006)
+        <https://doi.org/10.1051/0004-6361:20053818>`_ .
+
+        Parameters
+        ----------
+        bounding_box : tuple, optional
+            A pair of tuples, each consisting of two numbers
+            Represents the range of pixel values in both dimensions
+            ((xmin, xmax), (ymin, ymax))
+
+        bin_ext_name : str, optional
+            Extension name for the `~astropy.io.fits.BinTableHDU` extension.
+
+        crd_col_name : str, optional
+            Field name of the coordinate array in the structured array
+            stored in `~astropy.io.fits.BinTableHDU` data. This corresponds to
+            ``TTYPEi`` field in the FITS header of the binary table extension.
+
+        sampling : float, optional
+            The step, in input pixels, of the coordinate grid to be generated
+            for the coordinate array of the ``-TAB`` FITS WCS convention.
+
+        Returns
+        -------
+        hdr : astropy.io.fits.Header
+            Header with WCS-TAB information associated (to be used) with image
+            data.
+
+        bin_table : astropy.io.fits.BinTableHDU
+            Binary table extension containing the coordinate array.
+
+        Raises
+        ------
+        ValueError
+            When ``bounding_box`` is not defined either through the input
+            ``bounding_box`` parameter or this object's ``bounding_box``
+            property.
+
+        ValueError
```

Comment:
I'm confused by this description. It implies that the sampling parameter may be a list of numbers, one per axis, and that it is an error if not 1?

Reply:
Sampling can be a number or a list of numbers. If it is a number - it is assumed that the same sampling is applied to all axes. When the number of provided samplings is not 1, then the number of provided samplings **must** match the number of image's coordinate axes.

Reply:
I'll think about improving this description. Thanks.

Reply:
When reading the code, it was clear, but the docstring should say that it expects a sequence of sampling numbers and that a single one will be applied to all dimensions. Or something like that.